### DISCUSSION TITLE
2D meshing error for multiscale tessellations

### DISCUSSION THREAD

**Discussion by rjk5987** [neperfepx/neper#745] [Q&A] [answered]
Hello,

I've been working with @darrencpagan on 3D multiscale tessellations (grains subdivided into lamella) to achieve a target complex microstructure. After making significant progress in generating the desired tessellations, I have started attempting to mesh these, but the meshing consistently fails at 0% in the 2D meshing stage with the following error:
```
Error  : Wrong mesh dimension: -1!
```
To be clear, I have no issues meshing non-multiscale tessellations, but so far have not been able to successfully mesh a multiscale tessellation. I have tried varying the regularization of the tessellation and the rcl of the meshing, but the same error occurs regardless. As a minimal example, I will use the first example [here](https://neper.info/generation.html?#multiscale-tessellations). The commands I am using are:
```
export OMP_NUM_THREADS=1

neper -T -n 200::from_morpho -morpho "gg::lamellar(w=0.02)" -o "multiscale"

neper -M -elttype tet -rcl 0.9 multiscale.tess -o multiscale.xml
```
The tessellation finishes without issue. I have hidden its output below, if needed.

<details>
<summary>Expand for tessellation output.</summary>
<code>========================    N   e   p   e   r    =======================
Info   : A software package for polycrystal generation and meshing.
Info   : Version 4.6.1-1
Info   : Built with: gsl|muparser|opengjk|openmp|nlopt|libscotch (full)
Info   : Running on 1 threads.
Info   : <https://neper.info>
Info   : Copyright (C) 2003-2022, and GNU GPL'd, by Romain Quey.
Info   : No initialization file found (`/user/rjk5987/.neperrc').
Info   : ---------------------------------------------------------------
Info   : MODULE  -T loaded with arguments:
Info   : [ini file] (none)
Info   : [com line] -n 200::from_morpho -morpho gg::lamellar(w=0.02) -o
         multiscale
Info   : ---------------------------------------------------------------
Info   : Reading input data...
Info   : Creating domain...
Info   : Creating tessellations...
Info   :   [level] Running in multiscale mode (2 scales)...                           Info   :   [    1]  1/1 Info   :   - Setting seeds... 100%
Info   :   - Generating crystal orientations...
Info   :   - Running tessellation...
Info   :     > Initial solution: f   =2.165447574
Info   :     > Iteration  81321: fmin=0.000361053 f=0.000361053
Info   :     > Final solution  : f   =0.000361053 (81321 iterations)
Info   :     > Reached `eps' criterion.                                               Info   :   [    2]  1/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  2/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  3/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  4/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  5/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  6/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  7/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  8/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  9/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  10/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  11/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  12/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  13/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  14/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  15/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  16/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  17/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  18/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  19/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  20/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  21/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  22/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  23/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  24/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  25/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  26/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  27/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  28/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  29/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  30/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  31/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  32/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  33/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  34/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  35/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  36/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  37/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  38/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  39/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  40/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  41/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  42/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  43/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  44/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  45/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  46/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  47/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  48/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  49/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  50/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  51/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  52/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  53/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  54/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  55/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  56/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  57/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  58/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  59/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  60/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  61/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  62/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  63/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  64/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  65/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  66/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  67/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  68/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  69/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  70/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  71/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  72/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  73/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  74/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  75/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  76/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  77/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  78/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  79/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  80/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  81/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  82/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  83/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  84/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  85/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  86/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  87/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  88/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  89/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  90/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  91/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  92/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  93/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  94/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  95/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  96/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  97/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  98/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  99/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  100/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  101/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  102/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  103/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  104/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  105/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  106/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  107/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  108/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  109/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  110/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  111/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  112/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  113/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  114/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  115/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  116/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  117/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  118/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  119/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  120/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  121/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  122/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  123/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  124/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  125/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  126/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  127/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  128/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  129/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  130/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  131/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  132/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  133/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  134/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  135/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  136/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  137/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  138/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  139/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  140/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  141/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  142/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  143/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  144/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  145/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  146/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  147/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  148/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  149/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  150/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  151/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  152/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  153/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  154/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  155/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  156/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  157/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  158/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  159/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  160/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  161/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  162/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  163/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  164/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  165/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  166/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  167/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  168/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  169/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  170/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  171/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  172/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  173/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  174/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  175/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  176/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  177/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  178/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  179/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  180/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  181/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  182/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  183/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  184/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  185/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  186/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  187/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  188/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  189/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  190/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  191/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  192/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  193/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  194/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  195/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  196/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  197/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  198/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  199/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...                                       Info   :   [    2]  200/200 Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...
Info   : Flattening 2-scale tessellation...
Info   :   [level] Flattening 2-scale tessellation...                                 Info   :   [    1]  1/1: id=1
Info   : Writing results...
Info   :     [o] Writing file `multiscale.tess'...
Info   :     [o] Wrote file `multiscale.tess'.
Info   : Elapsed time: 249.518 secs.
========================================================================</code>
</details>

The meshing output is:
```
========================    N   e   p   e   r    =======================
Info   : A software package for polycrystal generation and meshing.
Info   : Version 4.6.1-1
Info   : Built with: gsl|muparser|opengjk|openmp|nlopt|libscotch (full)
Info   : Running on 1 threads.
Info   : <https://neper.info>
Info   : Copyright (C) 2003-2022, and GNU GPL'd, by Romain Quey.
Info   : No initialization file found (`/user/rjk5987/.neperrc').
Info   : ---------------------------------------------------------------
Info   : MODULE  -M loaded with arguments:
Info   : [ini file] (none)
Info   : [com line] -elttype tet -rcl 0.9 multiscale.tess -o
         multiscale.xml
Info   : ---------------------------------------------------------------
Info   : Reading input data...
Info   :   - Reading arguments...
Info   : Loading input data...
Info   :   - Loading tessellation...
Info   :     [i] Parsing file `multiscale.tess'...
Info   :     [i] Parsed file `multiscale.tess'.
Info   : Meshing...
Info   :   - Preparing... (cl = 0.03379) 100%
Info   :   - 0D meshing... 100%
Info   :   - 1D meshing... 100%
Info   :   - 2D meshing...   0% (0.58|0.7/100%| 0%| 0%)Error  : Wrong mesh dimension: -1!
Aborted (core dumped)
```
As I said before, all of my attempts at meshing a multiscale tessellation of this type (grains subdivided into lamella) fail with this error at 0% during 2D meshing, but I have no issues meshing non-multiscale tessellations.

Does anyone have an idea of what might be going wrong here? Thanks!!!

**Comment by rquey** [comment]
Well, if meshing works for non-multiscale tessellations, I don't know what happens.  This tessellation meshes fine for me:

```
neper -T -loadtess multiscale.tess -sel 0.002 -reg 1 -o multiscale-reg 
neper -M -elttype tet -rcl 0.9 -pl 1000 multiscale-reg.tess

========================    N   e   p   e   r    =======================
Info   : A software package for polycrystal generation and meshing.                                                                          
Info   : Version 4.6.1-8                                              
Info   : Built with: gsl|muparser|opengjk|openmp|nlopt|libscotch (full)
Info   : Running on 16 threads.                                                                                                              
Info   : <https://neper.info>                                                                                                                
Info   : Copyright (C) 2003-2022, and GNU GPL'd, by Romain Quey.                                                                             
Info   : Loading initialization file `/home/rquey/.neperrc'...        
Info   : ---------------------------------------------------------------                                                         
Info   : MODULE  -M loaded with arguments:                                                                                                   
Info   : [ini file] -tmp /home/rquey/tmp                                                                                                     
Info   : [com line] -elttype tet -rcl 0.9 -pl 1000 multiscale-reg.tess -o                                                                    
         multiscale.xml                                                                                                                      
Info   : ---------------------------------------------------------------
Info   : Reading input data...                                                                                                               
Info   :   - Reading arguments...                                     
Info   : Loading input data...                                        
Info   :   - Loading tessellation...                                  
Info   :     [i] Parsing file `multiscale-reg.tess'...                                                                                       
Info   :     [i] Parsed file `multiscale-reg.tess'.                                                                                          
Info   : Meshing...                                                                                                                          
Info   :   - Preparing... (cl = 0.03379) 100%                       
Info   :   - 0D meshing... 100%                                       
Info   :   - 1D meshing... 100%                                                                                                              
Info   :   - 2D meshing... 100% (0.00051|0.71/95%| 4%| 1%)                                                                                   
Info   :   - Fixing 2D-mesh pinches...                                                                                                       
Info   :   - 3D meshing... 100% (0.00014|0.66/91%| 9%| 0%)                                                                                                              
Info   : Searching nsets and fasets...
Info   : Writing mesh results...                                      
Info   :   - Preparing mesh...  
Info   :   - Mesh properties:                                         
Info   :     > Node number:   116740
Info   :     > Elt  number:   677853 
Info   :     > Mesh volume:    1.000 
Info   :   - Writing mesh...                                          
Info   :     [o] Writing file `multiscale.msh'...
Info   :     [o] Wrote file `multiscale.msh'.                         
Info   : Elapsed time: 567.246 secs.                                  
========================================================================

neper -V multiscale-reg.tess,multiscale.msh -dataelsetcol "scaleid(1)" -showelt1d all -dataelt1drad 0.003 -print img
```

![d](https://github.com/neperfepx/neper/assets/2659376/8c83efd4-3e60-436a-8ee8-31dbd71f7f39)


Are you running on Linux or Mac OS (as the `export OMP_NUM_THREADS=1` suggests)? It might be it.

(I'm on Ubuntu v22)

**Reply by mkasemer** [reply]
I think this is Ubuntu v22, though @darrencpagan and @rjk5987 can confirm.

Based on other discussion threads detailing a `-1` dimension error, this could be due to Gmsh. However, I am fairly certain that @darrencpagan followed my workflow for Gmsh installation from source, rather than installation via apt.

@darrencpagan, can you confirm: OS, Gmsh versions?

**Reply by rjk5987** [reply]
Hi Romain,

I am running on Linux (Ubuntu 20.04.6). I tried including ```export OMP_NUM_THREADS=1``` based on [this discussion](https://github.com/neperfepx/neper/discussions/493) to see whether it would make a difference, it did not. Interestingly, I ran the commands that you used,
```
neper -T -loadtess multiscale.tess -sel 0.002 -reg 1 -o multiscale-reg 
neper -M -elttype tet -rcl 0.9 -pl 1000 multiscale-reg.tess
```
and the 2D meshing is now reaching 19% before failing with the same error:
```
========================    N   e   p   e   r    =======================
Info   : A software package for polycrystal generation and meshing.
Info   : Version 4.6.1-1
Info   : Built with: gsl|muparser|opengjk|openmp|nlopt|libscotch (full)
Info   : Running on 80 threads.
Info   : <https://neper.info>
Info   : Copyright (C) 2003-2022, and GNU GPL'd, by Romain Quey.
Info   : No initialization file found (`/user/rjk5987/.neperrc').
Info   : ---------------------------------------------------------------
Info   : MODULE  -M loaded with arguments:
Info   : [ini file] (none)
Info   : [com line] -elttype tet -rcl 0.9 -pl 1000 multiscale-reg.tess
Info   : ---------------------------------------------------------------
Info   : Reading input data...
Info   :   - Reading arguments...
Info   : Loading input data...
Info   :   - Loading tessellation...
Info   :     [i] Parsing file `multiscale-reg.tess'...
Info   :     [i] Parsed file `multiscale-reg.tess'.
Info   : Meshing...
Info   :   - Preparing... (cl = 0.03379) 100%
Info   :   - 0D meshing... 100%
Info   :   - 1D meshing... 100%
Info   :   - 2D meshing...  19% (0.093|0.76/96%| 3%| 1%)Error  : Wrong mesh dimension: -1!
Aborted (core dumped)
```

**Reply by rquey** [reply]
Gmsh version?

**Reply by rjk5987** [reply]
4.5.1

**Reply by rquey** [reply, marked as answer]
It doesn't want to install on my laptop...

@rjk5987, it would be good idea to update your Gmsh version.

**Reply by rjk5987** [reply]
**Update**: We updated to Gmsh 4.11.1 (compiled from source) and it is now meshing without issue:
![img](https://github.com/neperfepx/neper/assets/136096866/9a456924-acd2-42cf-bf7a-38516b1e15d2)
I will note that we are not 100% sure whether the issue was the Gmsh version, or that our previous Gmsh installation was compiled against an older version of Ubuntu than we are currently using. Either way, thank you!
